pkg/ruler/ruler_test.go

```diff
@@ -1208,6 +1211,63 @@ func TestRuler_NotifySyncRulesAsync_ShouldTriggerRulesSyncingOnAllRulersWhenEnab
 	}
 }
 
+func TestRuler_InitialSyncFails(t *testing.T) {
```

Comment:
technically we're not testing that it retries. is it difficult to add that test too?